test/e2e/networking.go

```diff
@@ -101,155 +100,227 @@ var _ = Describe("Networking", func() {
-		passed := false
+			Logf("launching pod per node.....")
+			podNames := LaunchNetTestPodPerNode(i, nettestVersion, f, nodes, svcname)
 
-		//once response OK, evaluate response body for pass/fail.
-		var body []byte
-		getDetails := func() ([]byte, error) {
-			return f.Client.Get().
-				Namespace(f.Namespace.Name).
-				Prefix("proxy").
-				Resource("services").
-				Name(svc.Name).
-				Suffix("read").
-				DoRaw()
-		}
+			//The ugly debug logging makes it easy, by eye, to debug test results
+			//on a complex multithreaded run.
+			Logf("************************** Launched test pods for %v", i)
+			//By("Waiting for the webserver pods to transition to Running state")
 
-		getStatus := func() ([]byte, error) {
-			return f.Client.Get().
-				Namespace(f.Namespace.Name).
-				Prefix("proxy").
-				Resource("services").
-				Name(svc.Name).
-				Suffix("status").
-				DoRaw()
-		}
+			for _, podName := range podNames {
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred())
+				By(fmt.Sprintf("Waiting for connectivity to be verified [ port =  %v ] ", i))
+				//once response OK, evaluate response body for pass/fail.
+				PollPeerStatus(sem, f, svc)
 
-		timeout := time.Now().Add(2 * time.Minute)
-		for i := 0; !passed && timeout.After(time.Now()); i++ {
-			time.Sleep(2 * time.Second)
-			Logf("About to make a proxy status call")
-			start := time.Now()
-			body, err = getStatus()
-			Logf("Proxy status call returned in %v", time.Since(start))
-			if err != nil {
-				Logf("Attempt %v: service/pod still starting. (error: '%v')", i, err)
-				continue
 			}
-			// Finally, we pass/fail the test based on if the container's response body, as to wether or not it was able to find peers.
-			switch {
-			case string(body) == "pass":
-				Logf("Passed on attempt %v. Cleaning up.", i)
-				passed = true
-			case string(body) == "running":
-				Logf("Attempt %v: test still running", i)
-			case string(body) == "fail":
-				if body, err = getDetails(); err != nil {
-					Failf("Failed on attempt %v. Cleaning up. Error reading details: %v", i, err)
-				} else {
-					Failf("Failed on attempt %v. Cleaning up. Details:\n%s", i, string(body))
-				}
-			case strings.Contains(string(body), "no endpoints available"):
-				Logf("Attempt %v: waiting on service/endpoints", i)
-			default:
-				Logf("Unexpected response:\n%s", body)
-			}
-		}
 
-		if !passed {
-			if body, err = getDetails(); err != nil {
-				Failf("Timed out. Cleaning up. Error reading details: %v", err)
-			} else {
-				Failf("Timed out. Cleaning up. Details:\n%s", string(body))
-			}
-		}
-		Expect(string(body)).To(Equal("pass"))
-	})
+			Logf("************************** Finished test pods for %v", i)
+		}(nodes)
+	}
+	//Expect(string(body)).To(Equal("pass"))
+	//now wait for the all X nettests to complete...
+	//By("Waiting for all connectivities to be verified")
+	for ii := range ports {
+		Logf("Now waiting on port %v", ports[ii])
+		<-sem
```

Comment:
its not a huge deal to me, so am open to sync.WaitGroup if it gates a merge though.  I'd prefer channels for the flexibility so that we can possibly send data along them .